packages/nouns-contracts/test/foundry/BurnUpgradeForkMainnetTest.t.sol

```diff
@@ -0,0 +1,264 @@
+            values,
+            signatures,
+            calldatas,
+            'Upgrade to Auction House V2 & timelock V3'
+        );
+        console.log('Proposed proposalId: %d', proposalId);
+    }
+
+    function voteAndExecuteProposal(uint256 proposalId) internal {
+        vm.roll(
+            block.number +
+                NOUNS_DAO_PROXY_MAINNET.votingDelay() +
+                NOUNS_DAO_PROXY_MAINNET.proposalUpdatablePeriodInBlocks() +
+                1
+        );
+        NOUNS_DAO_PROXY_MAINNET.castVote(proposalId, 1);
+
+        vm.roll(block.number + NOUNS_DAO_PROXY_MAINNET.votingPeriod() + 1);
+        NOUNS_DAO_PROXY_MAINNET.queue(proposalId);
+
+        vm.warp(block.timestamp + NOUNS_TIMELOCK_V2_MAINNET.delay());
+        NOUNS_DAO_PROXY_MAINNET.execute(proposalId);
+    }
+
+    function test_proxiesWereUpgraded() public {
+        assertEq(get1967Implementation(auctionHouseProxy), address(newAuctionHouseLogic), 'upgrade failed');
+        assertEq(
+            get1967Implementation(address(NOUNS_TIMELOCK_V2_MAINNET)),
+            address(executorV3Logic),
+            'timelock upgrade failed'
+        );
+    }
+
+    function test_burnerIsOffAtFirst() public {
+        assertGt(burner.nextBurnNounID(), 1000000);
+
+        vm.expectRevert(ExcessETHBurner.NotTimeToBurnYet.selector);
+        burner.burnExcessETH();
+    }
+
+    function test_burn() public {
+        uint256 proposalId = proposeToTurnOnBurn(880, 10, 20);
+        voteAndExecuteProposal(proposalId);
+
+        assertEq(burner.nextBurnNounID(), 880);
+        assertEq(burner.minNewNounsBetweenBurns(), 10);
+        assertEq(burner.numberOfPastAuctionsForMeanPrice(), 20);
+
+        NounsAuctionHouseV2 ah = NounsAuctionHouseV2(auctionHouseProxy);
+        vm.warp(ah.auction().endTime + 1);
+        ah.settleCurrentAndCreateNewAuction();
+
+        for (uint256 i; i < 10; i++) {
+            ah.createBid{ value: 20 ether }(ah.auction().nounId);
+            vm.warp(ah.auction().endTime + 1);
+            ah.settleCurrentAndCreateNewAuction();
+        }
+
+        vm.expectRevert(ExcessETHBurner.NotEnoughAuctionHistory.selector);
+        burner.burnExcessETH();
+
+        for (uint256 i; i < 10; i++) {
+            ah.createBid{ value: 30 ether }(ah.auction().nounId);
+            vm.warp(ah.auction().endTime + 1);
+            ah.settleCurrentAndCreateNewAuction();
+        }
+
+        assertEq(burner.meanAuctionPrice(), 25 ether);
+        assertEq(NOUNS_DAO_PROXY_MAINNET.adjustedTotalSupply(), 422);
+        assertEq(burner.expectedTreasuryValueInETH(), 10550 ether);
+        assertEq(burner.treasuryValueInETH(), 13899.047817579079479237 ether);
+
+        // treasury has less ETH than 13899 - 25 * 422
+        assertEq(address(NOUNS_TIMELOCK_V2_MAINNET).balance, 2813.062865210366418892 ether);
+
+        burner.burnExcessETH();
+
+        assertEq(address(NOUNS_TIMELOCK_V2_MAINNET).balance, 0);
+    }
+
+    function proposeToTurnOnBurn(
```

Comment:
@davidbrai maybe this should also become a script?

Reply:
@eladmallel I've decided not to make the proposals a script because I think we will make the proposal manually.
it's not a complex proposal like the V3 upgrade, so I think it's safe.
wdyt?